service/history/replication/stream_receiver.go

```diff
@@ -257,26 +326,97 @@ func (r *StreamReceiverImpl) processMessages(
 		if streamResp.Err != nil {
 			return streamResp.Err
 		}
-		tasks := r.taskConverter.Convert(
+		if err := r.validateAndSetReceiverMode(streamResp.Resp.GetMessages().Priority); err != nil {
+			r.logger.Error("ReplicationTask wrong receiver mode", tag.Error(err))
+			break // sender mode changed, exit loop and let stream reconnect to retry
```

Comment:
If we just break here, Will it retry to reconnect?

Reply:
Yes it will reconnect. The recv event loop will exit with no error. And [here](https://github.com/temporalio/temporal/blob/5087766b908c14d06080fa74306181f3cae14b46/service/history/replication/stream.go#L74) the for loop will exit. StreamReceiverMonitor will retry the stream.

Reply:
Looks like StreamReceiverMonitor only reconnect on `StreamError` ([code](https://github.com/temporalio/temporal/blob/5087766b908c14d06080fa74306181f3cae14b46/service/history/replication/stream.go#L81)). For other error, it will retry the eventloop.

Reply:
yeah, so when the this loop is break, the receiver event loop will exit with no error. so it hit this [case](https://github.com/temporalio/temporal/blob/5087766b908c14d06080fa74306181f3cae14b46/service/history/replication/stream.go#L77) and exit the infinite for loop and let stream reconnect.

Reply:
I see. I missed the streamStopper.